### PR TITLE
HIVE-26816 : Add Jenkins file for branch-3

### PR DESCRIPTION
**Pull request by amanraj2520**
HIVE-26816 : Add Jenkins file for branch-3

### What changes were proposed in this pull request?
Added Jenkins file for pull requests made to branch-3.

### Why are the changes needed?
To validate that the changes people are making to branch-3 do not break.

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ * limitations under the License.
+ */
+
+properties([
+    // max 5 build/branch/day
+    rateLimitBuilds(throttle: [count: 5, durationName: 'day', userBoost: true]),
+    // do not run multiple testruns on the same branch
+    disableConcurrentBuilds(),
+    parameters([
+        string(name: 'SPLIT', defaultValue: '20', description: 'Number of buckets to split tests into.'),
```
Comment (cnauroth) [inline Jenkinsfile:25]: There was a recent change to increase splits to 22 on master. I'm wondering if that's relevant here for branch-3. CC: @ayushtkn .

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+      sh '''#!/bin/bash -e
+ls -l
+set -x
+. /etc/profile.d/confs.sh
+export USER="`whoami`"
+export MAVEN_OPTS="-Xmx2g"
+export -n HIVE_CONF_DIR
+cp $SETTINGS .git/settings.xml
+OPTS=" -s $PWD/.git/settings.xml -B -Dtest.groups= "
+OPTS+=" -Pitests,qsplits,dist"
```
Comment (cnauroth) [inline Jenkinsfile:92]: The `iceberg` and `errorProne` profiles are present on branch-3. Include them here?

**Review by cnauroth — CHANGES_REQUESTED** [review]
Thank you for setting this up, @amanraj2520 . I entered a few comments. Additionally, I see that master has some logic for repeating tests for all the supported metastore databases:

https://github.com/apache/hive/blob/master/Jenkinsfile#L275

Do we want to consider including that?

File: Jenkinsfile (hunk 1 above)
Comment (ayushtkn) [reply Jenkinsfile:25]: The number of tests in master will be way more than what are present in branch-3, so shouldn't be very much required here

File: Jenkinsfile (hunk 2 above)
Comment (ayushtkn) [reply Jenkinsfile:92]: Iceberg code was checked in in 4.0-alpha-1 [HIVE-24958](https://issues.apache.org/jira/browse/HIVE-24958)
I couldn't find the code here in branch-3
https://github.com/apache/hive/tree/branch-3

File: Jenkinsfile (hunk 1 above)
Comment (amanraj2520) [reply Jenkinsfile:25]: Yeah that is true

File: Jenkinsfile (hunk 2 above)
Comment (cnauroth) [reply Jenkinsfile:92]: Sorry, this was my mistake, looking at a bad local checkout of branch-3.

**Comment by amanraj2520** [comment]
@ayushtkn @cnauroth Then are we good to approve this? Please let me know if you need anything else. This is blocking cherry picks for 3.2.0 release so was considering merging it by today so that we can start development from tomorrow.

**Comment by amanraj2520** [comment]
Also just to add branch-3 is quite similar to branch-3.1 so I think this Jenkins file should be enough.

**Review by cnauroth — APPROVED** [review]
Thanks for responding to the feedback, @amanraj2520 . This is good from my perspective. +1 (non-binding). I do think testing with multiple metastore databases could be valuable, but that's not there in branch-3.1 either, and we could always decide to add it later in a separate patch.

**Comment by ayushtkn** [comment]
Nothing to add from my side either, Will let people tracking the 3.x release merge it :-)

**Comment by amanraj2520** [comment]
@nrg4878 @sunchao Can you try merging this PR. I do not have access to do that

**Review by zabetak — APPROVED** [review]
I will merge this PR now but please prioritize the stabilization of the build. Long running and failing tests can also lead to increased cloud costs that would be better to avoid.

**Comment by amanraj2520** [comment]
@zabetak Can you please mark this PR as merged and not closed

**Comment by zabetak** [comment]
@amanraj2520 Nope its not possible to do this. Closed does mean that it is not merged and its a common way of resolving PRs in this and other projects as well.

**Comment by amanraj2520** [comment]
@zabetak Sure no worries. I just had one question, (just a clarification) for all the merged PR's that I have raised for branch-3 let's say https://github.com/apache/hive/pull/3870 I see them as merged. Is it because they were merged after the Jenkins pipeline setup on branch-3.

**Comment by zabetak** [comment]
The "Merged" status appears when someone uses the respective UI button in the PR. When a PR is merged via the command line (as I usually do) it appears as "Closed".

**Comment by amanraj2520** [comment]
Got it @zabetak makes sense. Thanks

**Pull request closed without merging**